### DISCUSSION TITLE
Predefined Colors

### DISCUSSION THREAD

**Discussion by Geams** [bevyengine/bevy#4679] [Ideas]
Currently, the color constants defined in `bevy::render::color::Color` are not fully based on [CSS Color Module Level 3](https://www.w3.org/TR/css-color-3/#html4) but somewhere in between _section 4.1. Basic color keywords_ and _section 4.3. Extended color keywords_.
~~It could also be useful having the names available for the color constants, e.g. to display a tooltip with selecting from a color palette.~~

**Comment by DJMcNab** [comment]
The colours were originally added in https://github.com/bevyengine/bevy/pull/859, and it's unclear why the author chose the set they did.  Yeah, it's definitely on the todo list to update them to a full set.

I'm unclear what your second request is asking for. Is it a `HashMap<String, Color>`? That could be useful, although at that point I'd expect it to be provided by an external crate.

If not, https://github.com/bevyengine/bevy/pull/4387 might have already implemented what you wanted - the result is visible here <http://dev-docs.bevyengine.org/bevy/prelude/enum.Color.html#impl>.

**Reply by Geams** [reply]
Providing names with predefined colors is probably not really needed.
I simply use the `bevy::core::Name` component with a custom bundle to  display a tooltip.
That way, the user could also define own color names.
```
#[derive(Bundle)]
pub struct ColorBtnBundle{
    pub name: Name,
    #[bundle]
    pub button: ButtonBundle,
}
```
![color_tooltip](https://user-images.githubusercontent.com/35963028/167243430-80ecc324-25a4-4ee2-a484-e18de405b008.png)
